File: openrest4net/openrest4net-api/com/googlecode/openrest/CreditCard.cs
using System;

namespace com.googlecode.openrest
{
    class CreditCard
    {
        public CreditCard(string number, int expireMonth, int expireYear, string holderId,
            string holderName, bool anonymized)
        {
            this.number = number;
            this.expireMonth = expireMonth;
            this.expireYear = expireYear;
            this.holderId = holderId;
            this.holderName = holderName;
            this.anonymized = anonymized;
        }

        /** Empty constructor required for initialization from JSON-encoded string. */
        internal CreditCard() { }

        public string number;
        /** 1-based */
        public int? expireMonth;
        public int? expireYear;
        public string holderId;
        public string holderName;
        public bool? anonymized = false;
    }
}

File: trunk/openrest4j/openrest4j-api/src/com/googlecode/openrest/v1_1/CreditCard.java
package com.googlecode.openrest.v1_1;

import java.io.Serializable;

import org.codehaus.jackson.annotate.JsonIgnoreProperties;
import org.codehaus.jackson.map.annotate.JsonSerialize;

@JsonIgnoreProperties(ignoreUnknown = true)
public class CreditCard implements Serializable, Cloneable {
	public CreditCard(String number, Integer expireMonth, Integer expireYear,
            String holderId, String holderName, Boolean anonymized) {
        this.number = number;
        this.expireMonth = expireMonth;
        this.expireYear = expireYear;
        this.holderId = holderId;
        this.holderName = holderName;
        this.anonymized = anonymized;
    }

    /** Default constructor for JSON deserialization. */
    public CreditCard() {}
    
    @Override
	public Object clone() {
    	return new CreditCard(number, expireMonth, expireYear, holderId, holderName, anonymized);
	}

    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String number;

    /** 1-based */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public Integer expireMonth;

    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public Integer expireYear;

    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String holderId;

    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String holderName;

    @JsonSerialize(include = JsonSerialize.Inclusion.NON_DEFAULT)
    public Boolean anonymized = Boolean.FALSE;

    private static final long serialVersionUID = 1L;
}
